### ISSUE TITLE
W59 ("AWS::ApiGateway::Method should not have AuthorizationType set to 'NONE'.") should not trigger when HttpMethod = OPTIONS

### ISSUE THREAD

**Issue by lhgomes** [stelligent/cfn_nag#396]
W59 warning should not trigger to CORS Preflight HttpMethod (OPTIONS method). 

The CORS protocol requires the browser to send a preflight request to the server and wait for approval (or a request for credentials) from the server before sending the actual request.

Sample:
```
  apiOPTIONSC1907C21:
    Type: AWS::ApiGateway::Method
    Properties:
      HttpMethod: OPTIONS
      ResourceId:
        Fn::GetAtt:
          - api7B8C24E2
          - RootResourceId
      RestApiId:
        Ref: api7B8C24E2
      AuthorizationType: NONE
      Integration:
        IntegrationResponses:
          - ResponseParameters:
              method.response.header.Access-Control-Allow-Headers: "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,X-Amz-User-Agent'"
              method.response.header.Access-Control-Allow-Origin: "'*'"
              method.response.header.Access-Control-Allow-Methods: "'OPTIONS,GET,PUT,POST,DELETE,PATCH,HEAD'"
            StatusCode: "204"
        RequestTemplates:
          application/json: "{ statusCode: 200 }"
        Type: MOCK
      MethodResponses:
        - ResponseParameters:
            method.response.header.Access-Control-Allow-Headers: true
            method.response.header.Access-Control-Allow-Origin: true
            method.response.header.Access-Control-Allow-Methods: true
          StatusCode: "204"
```

**Comment by erickascic** [comment]
agreed.  we're on it

**Comment by erickascic** [comment]
fixed in 0.5.19
